compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -455,13 +466,99 @@ class Inliner(call: tpd.Tree, rhs: tpd.Tree)(implicit ctx: Context) {
       }
     }
 
-    override def typedApply(tree: untpd.Apply, pt: Type)(implicit ctx: Context) =
-      tree.asInstanceOf[tpd.Tree] match {
-        case Apply(Select(InlineableArg(closure(_, fn, _)), nme.apply), args) =>
-          inlining.println(i"reducing $tree with closure $fn")
-          typed(fn.appliedToArgs(args), pt)
-        case _ =>
-          super.typedApply(tree, pt)
+    override def typedApply(tree: untpd.Apply, pt: Type)(implicit ctx: Context) = {
+
+      def betaReduce(tree: Tree) = tree match {
```

Comment:
Would be nice to document this method since it's non-trivial and the name is a bit vague.

Reply:
I thought beta-reduce is as specific as it gets? 😄

Reply:
Just joking. I agree the details of this should be documented.